lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3OutputFile.java

```diff
@@ -50,10 +51,28 @@ public void createOrOverwrite(byte[] data)
         }
     }
 
+    @Override
+    public void createExclusive(byte[] data)
+            throws IOException
+    {
+        if (!context.exclusiveWriteSupported()) {
+            throw new UnsupportedOperationException("createExclusive not supported by " + getClass());
```

Comment:
```suggestion
throw new IllegalStateException("createExclusive operation is disabled");
```

Reply:
This is the same as default method

Reply:
We should not call `createExclusive` operation if the operation is not enabled - that's why i was arguing that IllegalStateException is better suited for such a situation.